src/cloudWatchLogs/registry/logStreamRegistry.ts

```diff
@@ -217,16 +220,18 @@ export async function filterLogEventsFromUriComponents(
                 events: response.events ? response.events : [],
                 nextForwardToken: response.nextToken,
                 nextBackwardToken: nextToken,
+                cancelled: false,
             }
         }
     } catch (err) {
-        if (!CancellationError.isUserCancelled(err)) {
-            getLogger().error(`cloudwatchlogs: failed to recieve log data for ${logGroupInfo}`)
-            throw err
-        } else {
+        if (CancellationError.isUserCancelled(err)) {
             getLogger().info('cloudwatchlogs: user cancelled search')
         }
     }
+    return {
```

Comment:
So should we still run through all the logic that can be run when an error occurs? What is the alternative to returning a `CloudWatchLogsResponse` with no events? I could also return `void` but that seemed to complicate things.

Reply:
> So should we still run through all the logic that can be run when an error occurs?

Should we? We didn't before, so we need to justify why we should do it now. It may help to start from the top of your program, maybe from `searchLogGroup`, and work your way down into where the AWS API call is made. Then assume that every single line can throw. Which they could. How does your program behave in these edge cases?

The entire purpose of an error is to short-circuit conventional control flow during _exceptional_ scenarios. Whether or not something is exceptional is a matter of perspective. From the user's perspective, something like a cancellation is obviously not an error. But we should not automatically assume the same is true for each component of our implementation.